internal/node/controlapi.go

```diff
@@ -421,16 +421,7 @@ func (api *ApiListener) handleDeploy(ctx context.Context, span trace.Span, m *na
 	}
 	span.AddEvent("Agent deploy request accepted")
 
-	if _, ok := api.mgr.handshakes[workloadID]; !ok {
```

Comment:
This seems like a necessary check, no? There could still be a situation that the handshake doesn't occur?